### PR TITLE
Fix support for executing benchmark on Windows

### PR DESCRIPTION
**Pull request by kibertoad**
#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

This PR fixes two problems with executing benchmarks:

1) `branch-comparer` depends on an old `inquirer` version that is not compatible with Windows. However unfortunate it is to create a fork for this purpose, chances of a project that didn't have updates for 3 years to fix it are rather slim (but I've submitted a PR: https://github.com/StarpTech/branch-comparer/pull/3)
2) Logger is unnecessarily enabled for performance benchmarking runs.

### CODE REVIEW

**Comment by StarpTech** [comment]
> However unfortunate it is to create a fork for this purpose, chances of a project that didn't have updates for 3 years to fix it are rather slim

When a library works for you, you don't need to update it :smile:

**Comment by kibertoad** [comment]
@StarpTech Oh, great news! If fixed library is released, I'll modify this pr to use the original again.

Now that we are on topic of changes, could you also add git repo to the package.json? I had to reverse engineer my way back to the repo from node_modules due to it not being there :D

**Comment by kibertoad** [comment]
@StarpTech Can you check if `fastify` benchmarking still works for you with the new version in this branch? It is completely broken on Windows currently (probably due to some of dependency updates), I wonder if it is specific to the OS or to fastify.

**Comment by StarpTech** [comment]
@kibertoad what's your definition of broken? It didn't work on the first run because we can't change to a remote branch when changes aren't committed. In 1.0.1 an error message is printed.

**Comment by kibertoad** [comment]
@StarpTech 
```
> branchcmp -r 2 -g -s "npm run benchmark:windows"

Current Branch is feat/windows-benchmark db19a72
? What's your branches to compare? feat/windows-benchmark, master
? Please enter the command you want to execute in all branches! npm run benchmark:windows
Checking out "feat/windows-benchmark"
Execute "npm run benchmark:windows"
events.js:187
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open 'C:\sources\fastify\branch.feat\windows-benchmark.round-1.log'
Emitted 'error' event on WriteStream instance at:
    at errorOrDestroy (internal/streams/destroy.js:108:12)
    at WriteStream.onerror (_stream_readable.js:759:7)
    at WriteStream.emit (events.js:210:5)
    at internal/fs/streams.js:294:12
    at FSReqCallback.oncomplete (fs.js:146:23) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path: 'C:\\sources\\fastify\\branch.feat\\windows-benchmark.round-1.log'
}
```

And when I compare to some older branch, I just get empty output without an error and then it quits with status code 0.

**Comment by kibertoad** [comment]
@StarpTech Tried with 1.0.1, now I get this:

```
Current Branch is feat/windows-benchmark 57b7dfc
? What's your branches to compare? master, remotes/master/master
? Please enter the command you want to execute in all branches! npm run benchmark:windows
Checking out "master"
Execute "npm run benchmark:windows"
Executed: "npm run benchmark:windows" and exited with code: 1
Execute "npm run benchmark:windows"
Executed: "npm run benchmark:windows" and exited with code: 1
Checking out "remotes/master/master"
Execute "npm run benchmark:windows"
events.js:187
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open 'C:\sources\fastify\branch.remotes\master\master.round-1.log'
Emitted 'error' event on WriteStream instance at:
    at errorOrDestroy (internal/streams/destroy.js:108:12)
    at WriteStream.onerror (_stream_readable.js:759:7)
    at WriteStream.emit (events.js:210:5)
    at internal/fs/streams.js:294:12
    at FSReqCallback.oncomplete (fs.js:146:23) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path: 'C:\\sources\\fastify\\branch.remotes\\master\\master.round-1.log'
}
```

**Comment by kibertoad** [comment]
@StarpTech Is there an mkdirp call somewhere that is supposed to happen? where do those directories come from?

**Comment by StarpTech** [comment]
Oh, I think I got it.

**Comment by StarpTech** [comment]
Try again with `1.0.2`

**Comment by kibertoad** [comment]
@StarpTech Sorry for the delay, was moving between apartments. Yes, that fixes the problem, thanks a lot! Works fine on Windows now.

File: package.json [hunk 1]
```diff
@@ -17,7 +17,9 @@
     "coverage": "npm run unit -- --cov --coverage-report=html",
     "test:ci": "npm run lint && npm run unit -- --cov --coverage-report=lcovonly && npm run test:typescript",
     "bench": "branchcmp -r 2 -g -s \"npm run benchmark\"",
+    "bench:windows": "branchcmp -r 2 -g -s \"npm run benchmark:windows\"",
```
Comment (StarpTech) [inline package.json:20]: Why do we need separate scripts for windows?

Comment (kibertoad) [reply package.json:20]: See `benchmark:windows`. When there is `npx` before `concurrently` on Windows, for some reason an error is thrown:
```
> npx concurrently -k -s first "node ./examples/simple.js" "npx autocannon -c 100 -d 5 -p 10 localhost:3000/"

10 '.' is not recognized as an internal or external command,
10 operable program or batch file.
10 The filename, directory name, or volume label syntax is incorrect.
10 ./examples/simple.js exited with code 1
```

If you have any ideas how to make script universally applicable (maybe some additional escaping is needed somewhere?..), that would be much better, of course.

Comment (StarpTech) [reply package.json:20]: Could you try

```
npx concurrently -k -s first "node './examples/simple.js'" "npx autocannon -c 100 -d 5 -p 10 localhost:3000/"
```

Comment (kibertoad) [reply package.json:20]: Same thing:
```
> npx concurrently -k -s first "node './examples/simple.js'" "npx autocannon -c 100 -d 5 -p 10 localhost:3000/"

10 '.' is not recognized as an internal or external command,
10 operable program or batch file.
10 The filename, directory name, or volume label syntax is incorrect.
```

. itself doesn't seem to be a problem, if I remove it, there is still an error:
```
> npx concurrently -k -s first "node 'examples/simple.js'" "npx autocannon -c 100 -d 5 -p 10 localhost:3000/"

10 'examples' is not recognized as an internal or external command,
10 operable program or batch file.
10 The filename, directory name, or volume label syntax is incorrect.
```

Comment (kibertoad) [reply package.json:20]: @StarpTech Do you have any other ideas, or should we keep it as-is?

Comment (Eomm) [reply package.json:20]: This works on my win10 with git bash (removed the apostrophe `'`)
```
npx concurrently -k -s first "node ./examples/simple.js" "npx autocannon -c 100 -d 5 -p 10 localhost:3000/"
```

Comment (kibertoad) [reply package.json:20]: It still doesn't for me for some reason:
image

**Comment by mcollina** [comment]
@Eomm are you ok for this to land?

**Comment by Eomm** [comment]
I would like to retry since it seems that we could just update the dep and avoid the `npx` usage to let it works on windows too, without defining a new script.

File: package.json (hunk 1 above)
Comment (Eomm) [reply package.json:20]: is the `npx` command working alone?

**Comment by mcollina** [comment]
@Eomm any update? We can also land this and then revisit.

**Comment by Eomm** [comment]
> @Eomm any update? We can also land this and then revisit.

Sorry for being late.

I have tried 2 win10 installation and both `benchmark:windows` and `benchmark` works.
The main issue was on `branch-comparer` for what I had the time to test

I would keep only one script `benchmark` script

**Comment by mcollina** [comment]
So this PR is not needed?

**Comment by kibertoad** [comment]
@mcollina Must be something on my end then. I will update PR to only bump the dependency,

**Comment by kibertoad** [comment]
@mcollina OK, it's ready and green now.

**Pull request merged**